packages/vkui/src/hooks/useTodayDate.ts

```diff
@@ -11,23 +12,55 @@ import { getMillisecondsToTomorrow } from '../lib/date';
  * @param listenDayChangesForUpdate - флаг по которому определяется, будет ли создаваться подписка на смену календарного дня
  */
 export function useTodayDate(listenDayChangesForUpdate = false) {
+  const { document } = useDOM();
   const [todayDate, setTodayDate] = React.useState(() => new Date());
 
   React.useEffect(() => {
-    if (!listenDayChangesForUpdate) {
+    if (!listenDayChangesForUpdate || !document) {
       return;
     }
 
-    const timeToDayChange = getMillisecondsToTomorrow(todayDate);
+    let timeout: NodeJS.Timeout | null = null;
 
-    const timeout = setTimeout(() => {
-      setTodayDate(new Date());
-    }, timeToDayChange);
+    const recalcTimeout = () => {
+      if (document.visibilityState === 'visible') {
+        const now = new Date();
+
+        const timeToDayChange = getMillisecondsToTomorrow(now);
+
+        // Удаляем старый таймаут
+        if (timeout) {
+          clearTimeout(timeout);
+        }
+
+        // Создаем новый таймаут
+        timeout = setTimeout(() => {
+          setTodayDate(now);
+        }, timeToDayChange);
+
+        // Если todayDate не обновился в таймаут - обновить при заходе на вкладку
+        if (!isSameDay(todayDate, now)) {
+          setTodayDate(now);
+        }
+      }
+    };
+
+    if (!timeout) {
+      recalcTimeout();
+    }
```

Comment:
`timeout` всегда `null`

```suggestion
    recalcTimeout();
```